### DISCUSSION TITLE
game crashes "null object reference" when stopping music on save data condition

### DISCUSSION THREAD

**Discussion by ReyingRexer** [HaxeFlixel/flixel#2612] [General]
- **Haxe version:** 4.2.5
- **Flixel version:** 4.11.0
- **OpenFL version:** ?.?.?
- **Lime version:** 7.9.0
- **Affected targets:**

________________________________

**Code snippet reproducing the issue**:

```haxe
package;

import flixel.FlxG;
import flixel.FlxState;
import flixel.system.FlxSound;
import flixel.system.debug.Window;
import flixel.text.FlxText;
import flixel.tweens.FlxEase;
import flixel.tweens.FlxTween;
import flixel.ui.FlxButton;
import flixel.util.FlxColor;
import flixel.util.FlxTimer;
import lime.app.Application;
import openfl.Lib;

class MenuState extends FlxState
{
	override public function create():Void
	{
		super.create();

		back = FlxG.sound.load(AssetPaths.back__ogg);
		back.volume = 0.3;
		back.persist = true;
		select = FlxG.sound.load(AssetPaths.select__ogg);
		select.volume = 0.3;
		select.persist = true;
		if (FlxG.save.data.music == true)
		{
			trace('turn it on');
			FlxG.sound.playMusic(AssetPaths.minecraft__ogg, 0.25, true);
		}
		else if (FlxG.save.data.music = = true)
		{
			trace('music was cancelled due to save');
			FlxG.sound.music.stop();
		}
		FlxG.camera.bgColor = 0x9400C3FF;
		FlxG.camera.flash(FlxColor.BLACK, 0.5);
		#if web
		var text = new FlxText(0, 0, 0, 'TWD Demo\nbrowser edition', 64);
		#end
		var text = new FlxText(0, 0, 0, 'TWD Demo', 64);
		text.screenCenter();
		add(text);
		var play = new FlxButton(188, 470, 'play', playingGame);
		add(play);
		var options = new FlxButton(729, 458, 'options', tooption);
		add(options);
	}
}
```

________________________________

**Observed behavior:on the save data condition, the false statement stops the music when no music detected**

**Expected behavior:crashes with message "null object reference"**

**Comment by Geokureli** [comment]
Converted to a discussion, since this seems like a logic issue with your code. If a flixel issue is discovered  we need to find out the specific cause before making an issue

What line is crashing? It should tell you, especially if you add `-debug` to your lime command. my guess is `FlxG.sound.music.stop();` perhaps there is no music instance to stop?

Also I notice you have the following
```hx
if (FlxG.save.data.music == true)
{}
else if (FlxG.save.data.music = = true)
```

`= =` wouldn't even compile due to the space in the middle, also it's the same condition as above. I assume you edited it in the issue from what was in your actual game.

**Reply by ReyingRexer** [reply]
Oh yeah, I forgot to tell you that even when I noticed that problem and fixed it, the issue would still apply.
and when I put a condition if (FlxG.sound.music.playing == true), it would still crash.

This is the new code

but it still crashes
```hx
		if (FlxG.save.data.music == true)
		{
			trace('turn it on');
			FlxG.sound.playMusic(AssetPaths.title__ogg, 0.25, true);
		}
		else if (FlxG.save.data.music == false)
		{
			trace('music was cancelled due to save');
			if (FlxG.sound.music.playing == true)
			{
				FlxG.sound.music.stop();
				FlxG.sound.music.volume = 0;
			}
		}
```

**Reply by Geokureli** [reply]
same question as before:
> What line is crashing? It should tell you, especially if you add -debug to your lime command. my guess is FlxG.sound.music.stop(); perhaps there is no music instance to stop?

**Reply by ReyingRexer** [reply]
Crashes at when it says if (FlxG.sound.music.playing == true)

**Reply by Geokureli** [reply]
So yeah, `FlxG.sound.music` is null until you call `FlxG.sound.playMusic`, you need to null check it before checking properties or calling methods on it

**Comment by ReyingRexer** [comment]
so your saying the condition should be if (FlxG.sound.music != null) then FlxG.sound.music.stop(); ?

**Reply by Geokureli** [reply]
yes

**Reply by ReyingRexer** [reply]
Ok, ill see if it works and ill notify you.

**Reply by ReyingRexer** [reply]
Update: it works, I really appreciate you helping me and Ill remember this chat, so that when I stumble upon this issue again I know what to do.
